### PR TITLE
Renci.runprop  merge request

### PR DESCRIPTION
**Pull request by BrianOBlanton**
moved run.properties messaging to output/transmit_rps.sh.  transmit_rps.sh now gets added as a post-processing script in config/YYYY config file

also added run.properties for forcing.metclass : synoptic|tropical

### CODE REVIEW

File: asgs_main.sh [hunk 1]
```diff
@@ -2843,8 +2850,8 @@ while [ true ]; do
       # nowcast directory; therefore, the non-existence of the nowcast
       # directory is evidence that something has gone wrong in prep
       if [[ ! -d $NOWCASTDIR ]]; then
-   	CURRENT_EVENT="REND"
-	   CURRENT_STATE="CMPL"
+   	     CURRENT_EVENT="REND"
```
Comment (wwlwpd) [inline asgs_main.sh:2853]: what's the purpose of this white space? They should lined up with everything else in the current block

File: asgs_main.sh [hunk 2]
```diff
@@ -3362,9 +3363,10 @@ while [ true ]; do
                   DATETIME=`date +'%Y-%h-%d-T%H:%M:%S%z'`
                   echo "time.post.start : $DATETIME" >> ${STORMDIR}/run.properties
                   scriptIndex=0
-                  while [[ $scriptIndex -lt ${#POSTPROCESS[@]} ]]; do
-                     #com="${OUTPUTDIR}/${POSTPROCESS[$scriptIndex]} $CONFIG $ADVISDIR $STORM $YEAR $ADVISORY $HPCENV $ENSTORM $CSDATE $HSTIME $GRIDFILE $OUTPUTDIR $SYSLOG $SSHKEY >> ${SYSLOG} 2>&1"
-                     com="${OUTPUTDIR}/${POSTPROCESS[$scriptIndex]} $CONFIG $ADVISDIR $STORM $YEAR $ADVISORY $HPCENV $ENSTORM $CSDATE $HSTIME $GRIDFILE $OUTPUTDIR $SYSLOG $SSHKEY "
+                  while [[ $scriptIndex -lt ${#POSTPROCESS[@]} ]]; do 
+                     com="${OUTPUTDIR}/${POSTPROCESS[$scriptIndex]} $CONFIG $ADVISDIR $STORM $YEAR $ADVISORY $HPCENV $ENSTORM $CSDATE $HSTIME $GRIDFILE $OUTPUTDIR $SYSLOG $SSHKEY >> ${SYSLOG} 2>&1"
+                     #com="${OUTPUTDIR}/${POSTPROCESS[$scriptIndex]} $CONFIG $ADVISDIR $STORM $YEAR $ADVISORY $HPCENV $ENSTORM $CSDATE $HSTIME $GRIDFILE $OUTPUTDIR $SYSLOG $SSHKEY "
```
Comment (wwlwpd) [inline asgs_main.sh:3368]: just delete this line that's commented out, there is no reason to keep it

File: asgs_main.sh [hunk 3]
```diff
@@ -3381,9 +3383,7 @@ while [ true ]; do
                fi
                CURRENT_EVENT="FORE"
                CURRENT_STATE="CMPL"
-   	         RMQMessage "INFO" "$CURRENT_EVENT" "$THIS>$ENSTORM" "$CURRENT_STATE" "Forecast Complete for Adv=$ADVISORY Ens=$ENSTORM"
-               # send current run.properties to RMQ
-               #RMQMessageRunProp "$STORMDIR"
+   	           RMQMessage "INFO" "$CURRENT_EVENT" "$THIS>$ENSTORM" "$CURRENT_STATE" "Forecast Complete for Adv=$ADVISORY Ens=$ENSTORM"
```
Comment (wwlwpd) [inline asgs_main.sh:3386]: spacing change only, can be moved over and eliminated from this diff it looks like

File: monitoring/logging.sh [hunk 4]
```diff
@@ -230,28 +231,18 @@ RMQMessageRunProp()  # send run.properties as a message to the asgs monitor queu
...
-     ${RMQMessaging_Script_RP} \
-         --Uid $$ \
-         --LocationName ${RMQMessaging_LocationName} \
-         --InstanceName $INSTANCENAME \
-         --Transmit ${RMQMessaging_Transmit} \
-         --input_filename "$RPDIR/run.properties" \
-         --output_filename "$RPDIR/run.properties.json" \
-          >> ${SYSLOG} 2>&1
-#   fi
+  #if [[ 10#$RMQADVISORY -lt 0 ]] ; then
```
Comment (wwlwpd) [inline monitoring/logging.sh:234]: delete commented out stuff, no reason to keep it

File: monitoring/logging.sh [hunk 5]
```diff
@@ -230,28 +231,18 @@ RMQMessageRunProp()  # send run.properties as a message to the asgs monitor queu
...
+  #  appMessage "warn: RMQA ($RMQADVISORY) < 0.  Not sending RunProp message ..." $APPLOGFILE
+  #	return
+  #fi
+  ${RMQMessaging_Script_RP} \
+     --Uid $$ \
+     --LocationName ${RMQMessaging_LocationName} \
+     --InstanceName $INSTANCENAME \
+     --Transmit ${RMQMessaging_Transmit} \
+     --input_filename "$RPDIR/run.properties" \
+     --output_filename "$RPDIR/run.properties.json" >> ${SYSLOG} 2>&1
```
Comment (wwlwpd) [inline monitoring/logging.sh:245]: any reason to check `$?` for an error?

File: monitoring/rp2json.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
```
Comment (wwlwpd) [inline monitoring/rp2json.py:1]: Assuming this requires python 2? There are no plans to introduce python 3 anytime soon. Just wanted to make sure and ask.

File: output/transmit_rps.sh [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+# this script can be called with just one command line option: the
+# full path to the run.properties file
+echo "Loading properties."
+# get loadProperties function
+SCRIPTDIR=`sed -n 's/[ ^]*$//;s/path.scriptdir\s*:\s*//p' $RUNPROPERTIES`
+source $SCRIPTDIR/properties.sh
+# load run.properties file into associative array
+loadProperties $RUNPROPERTIES
+#echo "Finished loading properties."   
+#CONFIG=${properties['config.file']}
```
Comment (wwlwpd) [inline output/transmit_rps.sh:41]: delete anything that's commented out, no reason to keep it

File: asgs_main.sh [hunk 8]
```diff
@@ -1670,16 +1670,16 @@ writeNAMProperties()
...
-   echo "forcing.nwp.year : ${ADVISORY:0:4}" >> $STORMDIR/run.properties
-   echo "forcing.nam.schedule.forecast.forecastcycle : \"${FORECASTCYCLE}\"" >> $STORMDIR/run.properties
-   echo "forcing.nwp.schedule.forecast.forecastselection : $forecastSelection" >> $STORMDIR/run.properties
-   echo "forcing.nam.forecast.download : $forecastDownload" >> $STORMDIR/run.properties
-   echo "forcing.nam.backsite : $BACKSITE" >> $STORMDIR/run.properties
-   echo "forcing.nam.backdir : $BACKDIR" >> $STORMDIR/run.properties
-   echo "forcing.nam.forecastlength : $FORECASTLENGTH" >> $STORMDIR/run.properties
-   echo "forcing.nam.reprojection.ptfile : $PTFILE" >> $STORMDIR/run.properties
-   echo "forcing.nam.local.altnamdir : $ALTNAMDIR" >> $STORMDIR/run.properties
+   echo "forcing.metclass : synoptic" >> $STORMDIR/run.properties
```
Comment (wwlwpd) [inline asgs_main.sh:1673]: trailing white space (example)

File: monitoring/rp2json.py [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+import os
+3import errno
```
Comment (wwlwpd) [inline monitoring/rp2json.py:4]: `3import` looks like a typo

**Review by wwlwpd — CHANGES_REQUESTED** [review]
All changes are required:

* do not remove existing keys from run.properties
* new keys need to be documented in the commit or change log (note: new keys only)
* roll this into a single commit
* ensure whitespace and commented-code issues are addressed
* need to know if any new python modules are required (and which ones)
* if new python modules are required, additional changes to support this need to be made in `cloud/general/init-python.sh`

I also need to know that this change works in a freshly built environment. So information on what testing was done would be helpful. I'd minimally expected:

* freshly built asgs (with latest `master` branch`)
* the `verify` command in `asgsh` runs successfully
* asgs was started w/monitoring options "on" and it worked as expected
* asgs was started w/monitoring options "off" and it worked as expected

Note: `monitoring/rp2json.py` had a typo in it that I'd expect to throw a runtime error; just happened to catch my eye. Please test this change thoroughly.

Thanks!

**Comment by jasonfleming** [comment]
Agree with not changing existing properties ... I am building a list of deprecated properties, but not removing them any time soon. We have had the writing of an official "ASGS Interface Guide" on the things-to-do list for years now to create and document a metadata standard for ASGS, but have not had the bandwidth to implement it. When converting to `run.properties` to `scenario.json` via `metadata.pl`, I don't carry the deprecated properties into the `json`. As @wwlwpd said, we need more coordinated action and focused effort to standardize our metadata.

**Pull request closed without merging**